### PR TITLE
KAFKA-4965: set internal.leave.group.on.close to false in StreamsConfig

### PR DESCRIPTION
**Pull request by dguy**
Set the internal consumer config internal.leave.group.on.close in `StreamsConfig`. This is to reduce the number of rebalances we get during bounces

### CODE REVIEW

**Comment by dguy** [comment]
@guozhangwang this is the follow up of: https://github.com/apache/kafka/pull/2650

**Comment by guozhangwang** [comment]
@dguy One question otherwise LGTM: are there any streams unit tests that can get affected by this change to have longer time to complete, e.g. reset integration test etc? If that is true should we override it back to true?

**Comment by dguy** [comment]
@guozhangwang yeah it might have an impact on any of the integration tests that use the same applicationId for multiple test methods. I'll update them to set it to true

**Comment by dguy** [comment]
retest this please

**Comment by dguy** [comment]
@ijuma if you get a chance can you please merge this?

File: streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java [hunk 1]
```diff
@@ -273,6 +273,7 @@ private Properties prepareTest() throws Exception {
         streamsConfiguration.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, 100);
         streamsConfiguration.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, "" + STREAMS_CONSUMER_TIMEOUT);
         streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        streamsConfiguration.put("internal.leave.group.on.close", true);
```
Comment (ijuma) [inline streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java]: Would it make sense to have a constant for this in some Streams internal utils class? It seems like a typo in any of these tests won't be caught by a test failure, right?

**Review by ijuma — COMMENTED** [review]
Thanks for the PR, left one comment. Also, do we have any Streams performance tests that we run regularly where we can see the impact of this change?

**Comment by dguy** [comment]
We have some benchmarks, but this will not impact them as they don't bounce any nodes. I'll run the streams system tests, though

**Review by ijuma — APPROVED** [review]
Thanks, LGTM.

Not a blocker for this PR, but it would be great to have some performance tests that validate all the work to improve performance when bouncing nodes or rebalancing.

**Pull request closed without merging**